server/src/main/java/io/crate/metadata/doc/DocIndexMetadata.java

```diff
@@ -643,7 +648,12 @@ private void createColumnDefinitions() {
     private Map<ColumnIdent, IndexReference> createIndexDefinitions() {
         MapBuilder<ColumnIdent, IndexReference> builder = MapBuilder.newMapBuilder();
         for (Map.Entry<ColumnIdent, IndexReference.Builder> entry : indicesBuilder.entrySet()) {
-            builder.put(entry.getKey(), entry.getValue().build(references));
+            var indexRef = entry.getValue().build(references);
+            if (indexRef.isDropped()) {
```

Comment:
Is this required, since we don't allow dropping indices?

Reply:
was going to add the same comment but afaik `create table (col string fulltext)` leads to `IndexReference` creation - it's a regular reference not a named index and can be dropped.

Reply:
Actually it isn't required as only named index definitions result in an `IndexReference`.

> `create table (col string fulltext)`

this isn't valid, the correct definitions would be `col text INDEX USING fulltext` which isn't a named index and such will result in a `SimpleReference`.

I will replace it with an assertion.

Reply:
>which isn't a named index and such will result in a SimpleReference.

I tried `testCreateTableWithInlineIndex` which has `create table quotes (quote string index using fulltext)`
quote is IndexReference. AFAIK, we don't use `SimpleReference` because we otherwise lose information about analyser.